writer.go

```diff
@@ -183,3 +156,51 @@ func trimPineconeKey(key string) (trimmed string, hasPrefix bool) {
 
 	return key, false
 }
+
+func parseRecords(ctx context.Context, records []sdk.Record) (upsertVecs []*pinecone.Vector, deleteIds []string, written int, err error) {
```

Comment:
Can we return the number of written records here at all, since this method is about parsing only?

Reply:
Oh yeah, sorry, the number is indeed unnecessary